swap_meet/vendor.py

```diff
@@ -1,2 +1,78 @@
 class Vendor:
-    pass
+    def __init__(self, inventory = None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+    
+    def add(self, item_to_add):
+        self.inventory.append(item_to_add)
+        return item_to_add
+
+    def remove(self, item_to_remove):
+        if item_to_remove not in self.inventory:
+            return False
+        self.inventory.remove(item_to_remove)
+        return item_to_remove
+
+    def get_by_category(self, category):
+        category_matches = []
+        for item in self.inventory:
+            if item.category == category:
+                category_matches.append(item)
+        return category_matches
+
+    def swap_items(self, other_vendor, selfs_item, others_item):
+        if selfs_item in self.inventory and others_item in other_vendor.inventory:
+            self.remove(selfs_item)
+            self.add(others_item)
+
+            other_vendor.remove(others_item)
+            other_vendor.add(selfs_item)
+
+            return True
+            
+        else:
+            return False
+    
+    def swap_first_item(self, other_vendor):
+        if (len(self.inventory) == 0) or (len(other_vendor.inventory) == 0):
+            return False
+        else:
```

Comment:
It's a little more pythonic to check for an empty list by taking advantage of its falsiness rather than explicitly checking the `len()`, as in

```python
        if not self.inventory or not other_vendor.inventory:
            return False
```

Also, notice that since the only thing you're doing in the positive condition is to `return`, this is essentially a guard clause. We can leave off the `else` and unindent the lines that were within it.